File: backend/pkg/domain/model/bookchapter.go
package model

import "github.com/gofrs/uuid"

type BookChapterID = uuid.UUID

type BookChapter struct {
	id     BookChapterID
	bookID BookID
	index  int
	title  string
}

func NewBookChapter(
	id BookChapterID,
	bookID BookID,
	index int,
	title string,
) BookChapter {
	return BookChapter{
		id:     id,
		bookID: bookID,
		index:  index,
		title:  title,
	}
}

func (bookChapter *BookChapter) ID() BookChapterID {
	return bookChapter.id
}

func (bookChapter *BookChapter) BookID() BookID {
	return bookChapter.bookID
}

func (bookChapter *BookChapter) Index() int {
	return bookChapter.index
}

func (bookChapter *BookChapter) Title() string {
	return bookChapter.title
}

func (bookChapter *BookChapter) SetTitle(title string) {
	bookChapter.title = title
}

File: frontend/src/shared/types/bookChapter.ts
export type BookChapter = {
  bookChapterId: string;
  index: number;
  title: string;
};

export type CreateBookChapter = {
  bookId: string;
  title: string;
};

export type EditBookChapter = {
  bookChapterId: string;
  title: string;
};

export type DeleteBookChapter = {
  bookChapterId: string;
};

export type ListBookChapter = {
  bookId: string;
};

export type ListBookChapterResponse = {
  bookChapters: BookChapter[];
};
